### DISCUSSION TITLE
How to correctly use union in moor?

### DISCUSSION THREAD

**Discussion by Ahmed-elshorbagy** [simolus3/drift#1423] [Q&A]
Hi
I have two tables with `DateTimeColumn` how to correctly make a union in moor that generate stream list and equals this `sql` statement

"
```
SELECT *, date1 AS savedate FROM table1 UNION 
    SELECT *, date2 AS savedate FROM table2  
ORDER BY savedate ASC

```
"
or i have to use only `customselectStream`
thanks in advance

**Comment by simolus3** [comment]
Hi! Unions are not supported by the Dart API at the moment. You can use a `customSelectStream`, but I strongly recommend using [queries validated at compile-time](https://moor.simonbinder.eu/docs/using-sql/custom_queries/#statements-with-a-generated-api). That also has the added advantage of moor generating a matching class for you.

**Reply by Ahmed-elshorbagy** [reply]
thanks but how can i use it correctly
my 
Table1 
has {
int userId
Date date1}
 Table2 has {
int userId
Date date2
}
my query is to union both table and sort by date and get rows with input userId
@simolus3

**Reply by Ahmed-elshorbagy** [reply]
I tried this

```
queries: {
    'unionTables':
'SELECT *, date_completed AS datesort FROM table1 WHERE user_id = ? UNION SELECT *, date AS datesort FROM table2 WHERE user_id = ? ORDER BY datesort ASC;'
  },
```

but it gives me this error 
`The sql query unionTables is invalid: line 1, column 1: The parts of this compound statement return different amount of columns`

**Reply by North101** [reply]
An SQL `UNION` needs to return the exact same amount of columns and the exact same data types for each column returned (matching the 1st `SELECT` in the `UNION`)

**Reply by Ahmed-elshorbagy** [reply]
thanks @North101
